pkg/thumbnails/storage/filesystem.go

```diff
@@ -0,0 +1,80 @@
+package storage
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/owncloud/ocis-thumbnails/pkg/config"
+)
+
+// NewFileSystemStorage creates a new instanz of FileSystem
+func NewFileSystemStorage(cfg config.FileSystemStorage) FileSystem {
+	return FileSystem{
+		dir: cfg.RootDirectory,
```

Comment:
Would this work on non UNIX systems?

Reply:
I'd leave UNIX by default and on a `Before` hook [check the host OS](https://golang.org/pkg/runtime/#pkg-constants) and update config